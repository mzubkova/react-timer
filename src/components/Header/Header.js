import React, { useContext } from "react";
import PropTypes from "prop-types";

import ButtonTheme from "../Button";

import { ThemeContext } from "../Context/ThemeContext";

import "./Header.scss";

const Header = () => {
  const { theme, setTheme } = useContext(ThemeContext);

  const toggleThemeMode = () => {
    setTheme(theme === "dark" ? "light" : "dark");
  };
  return (
    <div className={`header header--${theme}`}>
      <ButtonTheme></ButtonTheme>
    </div>
  );
};

Header.propTypes = {
  toggleThemeMode: PropTypes.func,
};

export default Header;
