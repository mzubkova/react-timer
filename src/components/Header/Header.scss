.header {
  padding: 10px 30px;
  height: 30px;
  width: auto;
  background-color: #cfcfcf;

  &--dark {
    background-color: #1f242d;
  }
}
