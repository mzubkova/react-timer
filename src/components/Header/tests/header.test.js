import React from "react";
import { shallow, mount } from "enzyme";
import Header from "../Header";

describe("Header", () => {
  it("Should match shapshot", () => {
    const component = shallow(<Header />);
    expect(component.html()).toMatchSnapshot();
  });
  it("Should have Header", () => {
    const component = mount(<Header />);
    expect(component.find("Header")).toHaveLength(1);
  });
  it("Should have div", () => {
    const component = mount(<Header />);
    expect(component.find("div")).toHaveLength(1);
  });
  it("Should have ButtonTheme", () => {
    const component = mount(<Header />);
    expect(component.find("ButtonTheme")).toHaveLength(1);
  });
});
