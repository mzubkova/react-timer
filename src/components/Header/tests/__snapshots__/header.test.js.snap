// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match shapshot 1`] = `"<div class=\\"header header--light\\"><button class=\\"switcher-btn\\">Dark theme</button></div>"`;
